### PR TITLE
Changes to make ROCK run on 16.04 machines

### PR DESCRIPTION
**Pull request by hemker**
I changed g2o to build from source. Furthermore, the buggy pcl osdep version for 16.04 is excluded to build it from source as well. Additionally, libprocps3 is not available  - changed to version 4

### CODE REVIEW

**Comment by jmachowinski** [comment]
please also remove the g2o-linker patch

**Comment by hemker** [comment]
what about the g2o_visodo_slam.patch?

**Comment by jmachowinski** [comment]
if it unused, also delete it...

File: rock.osdeps [hunk 1]
```diff
@@ -235,7 +235,7 @@ dc1394:
 
 vtk:
     debian,ubuntu:
-        '14.04,14.10,15.04,15.10':
+        '14.04,14.10,15.04,15.10,16.04':
```
Comment (doudou) [inline rock.osdeps]: This is undoing what others have just done, that is install vtk6 on 16.04.

File: rock.osdeps [hunk 2]
```diff
@@ -274,6 +274,7 @@ libproc:
     ubuntu:
         '10.04,10.10,11.04,11.10,12.04': libproc-dev
         '12.10,13.04,13.10': libprocps0-dev
+        '16.04': libprocps4-dev
```
Comment (doudou) [inline rock.osdeps]: How the version vs. default entries work is that the 'default' entry should point to the latest Ubuntu (as it is the most likely package name for version N+1). If the default changes, you have to add all the entries in-between, here:

```
'14.04,14.10,15.04,15.10': libprocps3-dev
default: libprocps4-dev # 16.04 and later ... until the next package name change
```

File: source.yml [hunk 3]
```diff
@@ -155,12 +155,10 @@ version_control:
             - $AUTOPROJ_SOURCE_DIR/patches/hogman_gcc_4.8.2.patch
     
     - slam/g2o:
-        type: svn
-        url: http://svn.openslam.org/data/svn/g2o/trunk
+        github: RainerKuemmerle/g2o.git
```
Comment (doudou) [inline source.yml]: You will have to warn the mailing list in advance for this, autobuild does not handle changing importers gracefully.

File: rock.osdeps (hunk 1 above)
Comment (hemker) [inline rock.osdeps]: But this conflicts with rock.core osdeps:
vtk-qt4:
  80     ubuntu:
  81       '16.04': [libvtk5-qt4-dev,libvtk5-dev]
  82       default: libvtk5-qt4-dev

Comment (doudou) [inline rock.osdeps]: Then you should probably revert the change (i.e. remove the 'default' entry and replace the 14.04 ... version list by default. And make sure that someone tried it on a clean install (i.e. a clean VM)

Comment (hemker) [inline rock.osdeps]: when changing the osdep to 'default: [libvtk5-qt4-dev,libvtk5-dev]' and keeping the use of libvtk6-dev in rock osdeps, it conflicts with the installation of python-vtk6.
I purged all libvtk* before

Comment (hemker) [inline rock.osdeps]: seems to work when changing the default python-vtk version to python-vtk6 in rock/rock.osdeps

**Comment by doudou** [comment]
I've cherry-picked the procps change.

For the vtk change, I would like to make sure we have something coherent between rock.core and rock

For g2o, I would prefer having the opinion of g2o users ...

File: rock.osdeps (hunk 1 above)
Comment (planthaber) [inline rock.osdeps]: libvtk5-qt4-dev just became unavailable in debian testing.
https://github.com/rock-core/package_set/pull/90/files

Afaik the dependency that adds vtk-qt is the rock-widget-collection, i don't mind changing the rock.core osdep because i just fixed the vtk-qt4 one 10 days ago:

https://github.com/rock-core/package_set/pull/84/files

**Comment by jmachowinski** [comment]
all our slam guys moved to the github: RainerKuemmerle/g2o repo.
So it sounds sane to do this upstream.

**Comment by jmachowinski** [comment]
All patches are now in master
